lib/appsignal/rack/body_wrapper.rb

```diff
@@ -72,6 +72,18 @@ def method_missing(method_name, *args, &block)
         @body.__send__(method_name, *args, &block)
       end
       ruby2_keywords(:method_missing) if respond_to?(:ruby2_keywords, true)
+
+      private
+
+      def appsignal_report_error(error)
+        @transaction.set_error(error) if appsignal_accepted_error?(error)
+      end
+
+      def appsignal_accepted_error?(error)
+        return true unless error.cause
+
+        !IGNORED_ERRORS.include?(error.cause.class)
```

Comment:
Not sure if this is necessary, but an error cause can in turn have another error cause. We could traverse these to see if any of the causes is ultimately an ignored error.

```suggestion
        return false if IGNORED_ERRORS.include?(error.cause.class)
        appsignal_accepted_error?(error.cause)
```